### PR TITLE
Update the canary examples so it can send different versions of TriggeredVX

### PR DESCRIPTION
**Pull request by tommpy**

### CODE REVIEW

File: src/main/scala/com/ovoenergy/comms/example/Main.scala [hunk 1]
```diff
@@ -4,48 +4,74 @@ import java.time.Instant
...
-                                         preferredChannels: Option[List[Channel]]): Future[_] = {
-    val triggeredEvent = TriggeredEvent.buildEvent(traceToken, scheduleFor, preferredChannels)
-    println(s"Sending Triggered event with traceToken $traceToken")
-    Kafka.send(triggeredEvent)
+  implicit val triggeredV3Sender = new TriggerSender[TriggeredV3] {
+    override def apply(t: TriggeredV3): Future[RecordMetadata] = Kafka send t
   }
+}
 
+object Main {
```
Comment (cb372) [inline src/main/scala/com/ovoenergy/comms/example/Main.scala]: Not sure that the Lambda will still work now you've moved the entry point from a class to an object

File: src/main/scala/com/ovoenergy/comms/example/TriggeredEventV2.scala [hunk 2]
```diff
@@ -5,9 +5,9 @@ import com.ovoenergy.comms.model._
 import java.time.{Instant, OffsetDateTime}
 import java.util.UUID
 
-object TriggeredEvent {
+object TriggeredEventV2 {
 
-  @Template("service", "canary", "4.0")
+  @Template("service", "canary", "3.0")
```
Comment (cb372) [inline src/main/scala/com/ovoenergy/comms/example/TriggeredEventV2.scala]: why the downgrade from 4.0 to 3.0?

File: src/test/scala/com/ovoenergy/comms/example/AcceptanceSpec.scala [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+class AcceptanceSpec extends FlatSpec with BeforeAndAfterAll with EmbeddedKafka with MessageHelper with Matchers {
+  private val topicV2 = "comms.triggered.v2"
+  private val topicV3 = "comms.triggered.v3"
+
+  override protected def beforeAll(): Unit = {
+    EmbeddedKafka.start()
+    createCustomTopic(topicV2)
+    createCustomTopic(topicV3)
+    sys.env
```
Comment (cb372) [inline src/test/scala/com/ovoenergy/comms/example/AcceptanceSpec.scala]: ?

File: src/test/scala/com/ovoenergy/comms/example/AcceptanceSpec.scala [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+    triggered.get.metadata should have (
+      'traceToken ("someTraceToken"),
+      'deliverTo (ContactDetails(Some("dilbert@path-e-tech-management.com"), None)),
+      'commManifest (CommManifest(Service, "canary", "3.0"))
+    )
+  }
+}
+
+//Right(List(MessageV2(traceToken1,None,None,None,None), MessageV2(traceToken2,Some(123),None,None,None), MessageV2(traceToken3,Some(123),Some(List(SMS, Email)),None,None), MessageV2(traceToken4,Some(123),None,Some(some@email.com),None), MessageV2(traceToken5,Some(123),None,None,Some(+01234567890))))
```
Comment (cb372) [inline src/test/scala/com/ovoenergy/comms/example/AcceptanceSpec.scala]: delete this garbage?

File: src/test/scala/com/ovoenergy/comms/example/SNSSpec.scala [hunk 5]
```diff
@@ -1,53 +1,53 @@
...
+        SendCanaryV2("traceToken1", None, None, None, None),
+        SendCanaryV2("traceToken2", Some(123), None, None, None),
+        SendCanaryV2("traceToken3", Some(123), Some(List("SMS", "Email")), None, None),
+        SendCanaryV2("traceToken4", Some(123), None, Some("some@email.com"), None),
+        SendCanaryV2("traceToken5", Some(123), None, None, Some("+01234567890"))
+      )))
+  }
 }
+
+//Right(List(MessageV1(traceToken1,None,None), MessageV1(traceToken2,Some(123),None), MessageV1(traceToken3,Some(123),Some(List(SMS, Email))), MessageV1(traceToken4,Some(123),None), MessageV1(traceToken5,Some(123),None)))
```
Comment (cb372) [inline src/test/scala/com/ovoenergy/comms/example/SNSSpec.scala]: same here

File: src/test/scala/com/ovoenergy/comms/example/AcceptanceSpec.scala (hunk 3 above)
Comment (tommpy) [inline src/test/scala/com/ovoenergy/comms/example/AcceptanceSpec.scala]: Good spot

File: src/main/scala/com/ovoenergy/comms/example/TriggeredEventV2.scala (hunk 2 above)
Comment (tommpy) [inline src/main/scala/com/ovoenergy/comms/example/TriggeredEventV2.scala]: Bad merge I think

File: src/test/scala/com/ovoenergy/comms/example/AcceptanceSpec.scala (hunk 4 above)
Comment (tommpy) [inline src/test/scala/com/ovoenergy/comms/example/AcceptanceSpec.scala]: Yup

File: src/test/scala/com/ovoenergy/comms/example/SNSSpec.scala (hunk 5 above)
Comment (tommpy) [inline src/test/scala/com/ovoenergy/comms/example/SNSSpec.scala]: Gone.

**Pull request merged**